control-plane/connect-inject/webhook/mesh_webhook_test.go

```diff
@@ -1129,6 +1132,231 @@ func TestHandlerHandle(t *testing.T) {
+				},
+				{
+					Operation: "remove",
+					Path:      "/spec/containers/0/readinessProbe/httpGet",
+				},
+				{
+					Operation: "add",
+					Path:      "/spec/containers/0/securityContext",
+				},
+				{
+					Operation: "remove",
+					Path:      "/spec/containers/0/startupProbe",
+				},
+				{
+					Operation: "remove",
+					Path:      "/spec/containers/0/livenessProbe",
+				},
+				{
+					Operation: "add",
+					Path:      "/metadata/labels",
+				},
+				{
+					Operation: "add",
+					Path:      "/metadata/annotations/" + escapeJSONPointer(constants.KeyInjectStatus),
+				},
+				{
+					Operation: "add",
+					Path:      "/metadata/annotations/" + escapeJSONPointer(constants.KeyTransparentProxyStatus),
+				},
+
+				{
+					Operation: "add",
+					Path:      "/metadata/annotations/" + escapeJSONPointer(constants.AnnotationOriginalPod),
+				},
+				{
+					Operation: "add",
+					Path:      "/metadata/annotations/" + escapeJSONPointer(constants.AnnotationConsulK8sVersion),
+				},
+			},
+		},
+	}
+
+	for _, tt := range cases {
+		t.Run(tt.Name, func(t *testing.T) {
+			tt.Webhook.ConsulConfig = &consul.Config{HTTPPort: 8500}
+			ctx := context.Background()
+			resp := tt.Webhook.Handle(ctx, tt.Req)
+			if (tt.Err == "") != resp.Allowed {
+				t.Fatalf("allowed: %v, expected err: %v", resp.Allowed, tt.Err)
+			}
+			if tt.Err != "" {
+				require.Contains(t, resp.Result.Message, tt.Err)
+				return
+			}
+
+			var iptablesCfg iptables.Config
+			var overwritePorts []string
+			actual := resp.Patches
+			if len(actual) > 0 {
+				for i := range actual {
+					if actual[i].Path == "/spec/initContainers" {
+						value, ok := actual[i].Value.([]any)
+						require.True(t, ok)
+
+						valueMap, ok := value[0].(map[string]any)
+						require.True(t, ok)
+
+						envs, ok := valueMap["env"].([]any)
+						require.True(t, ok)
+
+						redirectEnv, ok := envs[8].(map[string]any)
+						require.True(t, ok)
+
+						require.Equal(t, redirectEnv["name"].(string), "CONSUL_REDIRECT_TRAFFIC_CONFIG")
+						iptablesJson, ok := redirectEnv["value"].(string)
+						require.True(t, ok)
+
+						err := json.Unmarshal([]byte(iptablesJson), &iptablesCfg)
+						require.NoError(t, err)
+					}
+
+					if strings.Contains(actual[i].Path, "/spec/containers/1") {
+						valueMap, ok := actual[i].Value.(map[string]any)
+						require.True(t, ok)
+
+						for k, v := range valueMap {
+							if strings.Contains(k, "Probe") {
+								probe, ok := v.(map[string]any)
+								require.True(t, ok)
+
+								httpProbe, ok := probe["httpGet"]
+								require.True(t, ok)
+
+								httpProbeMap, ok := httpProbe.(map[string]any)
+								require.True(t, ok)
+
+								port, ok := httpProbeMap["port"]
+								require.True(t, ok)
+
+								portNum, ok := port.(float64)
+								require.True(t, ok)
+
+								overwritePorts = append(overwritePorts, strconv.Itoa(int(portNum)))
+							}
+						}
+					}
+
+					actual[i].Value = nil
```

Comment:
I'm not 100% clear on this step, would it be useful to add a comment explaining the need to set the Value to nil?

Reply:
Yeah, it's another copy-paste artifact. I also found this hard to understand. We don't actually validate the output is correct, just that the right keys change. I will add a comment though.